### PR TITLE
refactor array_agg to not to have `update` and `merge`

### PR DESCRIPTION
**Pull request by Jimexist**
# Which issue does this PR close?

Closes #.

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by Jimexist** [comment]
related #1598

File: datafusion/src/physical_plan/expressions/array_agg.rs [hunk 1]
```diff
@@ -95,77 +95,60 @@ impl AggregateExpr for ArrayAgg {
...
         if values.is_empty() {
             return Ok(());
         };
-        (0..values[0].len()).try_for_each(|index| {
-            let v = values
-                .iter()
-                .map(|array| ScalarValue::try_from_array(array, index))
-                .collect::<Result<Vec<_>>>()?;
-            self.update(&v)
+        assert!(values.len() == 1, "array_agg can only take 1 param!");
```
Comment (alamb) [inline datafusion/src/physical_plan/expressions/array_agg.rs:117]: I like the new assert 👍

**Review by alamb — APPROVED** [review]
Looks like a nice cleanup to me 👍 

FYI @viirya  (who implemented this originally in https://github.com/apache/arrow-datafusion/pull/1300 I think)

**Comment by alamb** [comment]
I think the API change was in another PR https://github.com/apache/arrow-datafusion/pull/1582

**Pull request merged**